### PR TITLE
play kube: fix segfault

### PR DESCRIPTION
**Pull request by haircommander**
when securityContext wasn't specified in yaml.
add a test as well

fixes: https://github.com/containers/libpod/issues/3956

Signed-off-by: Peter Hunt <pehunt@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
Sure, LGTM

**Comment by mheon** [comment]
@rhatdan @baude @vrothberg @giuseppe @TomSweeneyRedHat PTAL

File: pkg/adapter/pods.go [hunk 1]
```diff
@@ -683,25 +683,27 @@ func kubeContainerToCreateConfig(ctx context.Context, containerYAML v1.Container
...
 	}
 
-	if containerConfig.SecurityOpts != nil {
-		if containerYAML.SecurityContext.ReadOnlyRootFilesystem != nil {
-			containerConfig.ReadOnlyRootfs = *containerYAML.SecurityContext.ReadOnlyRootFilesystem
-		}
-		if containerYAML.SecurityContext.Privileged != nil {
-			containerConfig.Privileged = *containerYAML.SecurityContext.Privileged
-		}
+	if containerYAML.SecurityContext != nil {
```
Comment (rhatdan) [inline pkg/adapter/pods.go]: Could have made this a lot simpler by combining these two if commands on same line.

Comment (haircommander) [reply pkg/adapter/pods.go]: for some reason, the AllowPRivilegeEscalation check was outside of the containerConfig.SecurityOpts check, so doing it this way maintained the branch structure. I didn't know enough about why that happened to choose to change it

**Pull request merged**